docs/design_proposals/multi_environment_proposal.md

```diff
@@ -0,0 +1,277 @@
+# Proposal Design: Multi Environment Support
+## Objective
+The aim is to introduce an environment set mechanism in the `pixi` package manager.
+This mechanism will enable clear, conflict-free management of dependencies tailored to specific environments, while also maintaining the integrity of fixed lockfiles.
+
+
+### Motivating Example
+There are multiple scenarios where multiple environments are useful.
+
+- **Testing of multiple package versions**, e.g. `py39` and `py310` or polars `0.12` and `0.13`.
+- **Smaller single tool environments**, e.g. `lint` or `docs`.
+- **Large developer environments**, that combine all the smaller environments, e.g. `dev`.
+- **Strict supersets of environments**, e.g. `prod` and `test-prod` where `test-prod` is a strict superset of `prod`.
+- **Multiple machines from one project**, e.g. a `cuda` environment and a `cpu` environment.
+- **And many more.** (If you have a use-case please add it to the list, so we can make sure it's covered)
```

Comment:
```suggestion
- **And many more.**
```
Probably don't want that last part, right? That was just while we were editing on HackMD.

Reply:
I was actually planning to leave that in to motivate readers to help with more use cases. At some point this will probably move into actual feature documentation but until that, this is probably a good place to keep the use cases. 

Maybe this is more clear:
```suggestion
- **And many more.** (Feel free to edit this document in our GitHub and add your use case.)
```

Reply:
Yeah, that sounds clearer.